src/Swift.Runtime/src/Swift/SwiftArray.cs

```diff
@@ -121,17 +121,22 @@ static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescript
     /// <summary>
     /// Constructs a new SwiftArray from the given handle.
     /// </summary>
-    unsafe SwiftArray(SwiftHandle handle)
+    unsafe SwiftArray(IntPtr handle)
     {
-        _buffer = *(SwiftHandle*)handle;
+        IntPtr bufferPtr = (IntPtr)NativeMemory.Alloc((nuint)sizeof(IntPtr));
+        System.Buffer.MemoryCopy((void*)handle, (void*)bufferPtr, sizeof(IntPtr), sizeof(IntPtr));
```

Comment:
```suggestion
```suggestion
        IntPtr* bufferPtr = (IntPtr*)NativeMemory.Alloc((nuint)sizeof(IntPtr));
        *bufferPtr = result;
```
```